### PR TITLE
Remove broken symlink from node_modules on uninstall

### PR DESCRIPTION
**Pull request by panuhorsmalahti**
Partial fix for #1684.

We could improve this by also removing the symlink on install if the symlink exists. This PR will fix at least some of the cases, however. See issue for more details.

### CODE REVIEW

File: src/extensions/extension-discovery.ts [hunk 1]
```diff
@@ -208,12 +208,17 @@ export class ExtensionDiscovery {
...
    */
-  async uninstallExtension(absolutePath: string) {
-    logger.info(`${logModule} Uninstalling ${absolutePath}`);
+  async uninstallExtension({ absolutePath, manifest }: InstalledExtension) {
+    logger.info(`${logModule} Uninstalling ${manifest.name}`);
+
+    // remove the symlink under node_modules.
+    // If we don't remove the symlink, the uninstall would leave a non-working symlink,
+    // which wouldn't be fixed if the extension was reinstalled, causing the extension not to work.
+    await fs.remove(this.getInstalledPath(manifest.name));
```
Comment (jakolehm) [inline src/extensions/extension-discovery.ts:221]: Is there any possibility that symlink does not exist? Should we handle that case?

Comment (panuhorsmalahti) [reply src/extensions/extension-discovery.ts:221]: .remove does nothing if the path doesn't exist so this should be safe.

File: src/extensions/extension-discovery.ts [hunk 2]
```diff
@@ -260,6 +265,22 @@ export class ExtensionDiscovery {
...
+    return path.join(this.nodeModulesPath, name);
+  }
+
+  /**
+   * Returns the symlinked path to the package.json,
+   * e.g. "/Users/<username>/Library/Application Support/Lens/node_modules/@publisher/extension/package.json"
+   */
+  protected getInstalledManifestPath(name: string) {
+    return path.join(this.getInstalledPath(name), manifestFilename);
+  }
```
Comment (Nokel81) [inline src/extensions/extension-discovery.ts:282]: It is true that I don't understand the whole of our extensions install process. However, I would like to know why we have to care about symlinks at all. Why not just "require" the package from `~/.k8slens/extensions` without needing symlinks?

Comment (panuhorsmalahti) [reply src/extensions/extension-discovery.ts:282]: Probably for making it easy to install the dependencies. Making a big change to the extensions npm stuff will be a separate task in any case.

Comment (Nokel81) [reply src/extensions/extension-discovery.ts:282]: of course

**Review by Nokel81 — APPROVED** [review]

**Review by jim-docker — COMMENTED** [review]
LGTM but I'm going to give it some testing...

**Comment by jim-docker** [comment]
This may be an artificial test situation but:
If I install the extension, verify it works, delete the installed files via 
```
rm -Rf ~/.k8slens/extensions/mirantis--lens-extension-cc
```
then go to Lens extension page and uninstall the extension, I observe 
1) a notification pops up saying "Error uninstalling..."
2) that the no-longer-valid link has been removed, as expected.
3) extension is removed from the list, as expected

However, if instead of clicking "Uninstall" after doing the `rm -Rf ...` I click "Disable", I observe 
1) Lens disables the extension, 
2) after a few seconds the extension disappears from the list (automatically uninstalled by watcher?)
3) but the link has **not** been removed.

**Review by jakolehm — APPROVED** [review]

**Comment by panuhorsmalahti** [comment]
> This may be an artificial test situation but:
> If I install the extension, verify it works, delete the installed files via
> 
> ```
> rm -Rf ~/.k8slens/extensions/mirantis--lens-extension-cc
> ```
> 
> then go to Lens extension page and uninstall the extension, I observe
> 
> 1. a notification pops up saying "Error uninstalling..."
> 2. that the no-longer-valid link has been removed, as expected.
> 3. extension is removed from the list, as expected
> 
> However, if instead of clicking "Uninstall" after doing the `rm -Rf ...` I click "Disable", I observe
> 
> 1. Lens disables the extension,
> 2. after a few seconds the extension disappears from the list (automatically uninstalled by watcher?)
> 3. but the link has **not** been removed.

I believe that is working as intended. I can go through what is happening in more detail tomorrow. I think we can improve various error sitations with other PRs (I have few in mind).

**Pull request merged**